rest-api/dao/participant_dao.py

```diff
@@ -344,6 +344,34 @@ def add_missing_hpo_from_site(self, session, participant_id, site_id):
     # Update the version and add history row
     self._do_update(session, participant, participant)
 
+  def switch_to_test_account(self, session, participant_id):
+    test_hpo = HPODao().get_by_name(TEST_HPO_NAME)
+    if test_hpo is None:
+      raise BadRequest('No test hpo found for name ' + TEST_HPO_NAME)
+
+    test_hpo_id = test_hpo.hpoId
+    site = self.site_dao.get_by_hpo_id(test_hpo_id)
+    if site is None:
+      raise BadRequest('Invalid siteId reference hpo_id %r.' % test_hpo_id)
+
+    participant = self.get_for_update(session, participant_id)
+    if participant is None:
+      raise BadRequest('No participant %r for HPO ID udpate.' % participant_id)
+
+    if participant.hpoId == test_hpo_id:
+      return
+
+    participant.hpoId = site.hpoId
+    participant.organizationId = site.organizationId
+    participant.siteId = site.siteId
+    participant.providerLink = make_primary_provider_link_for_id(site.hpoId)
+    if participant.participantSummary is None:
+      raise RuntimeError('No ParticipantSummary available for P%d.' % participant_id)
```

Comment:
This wouldn't be a time to raise a RuntimeError. It's an inaccurate message. Also, since this function is only called based on a participant summary, they should always have one.

Reply:
This is a public function, we can't make sure it will be alway called based on a participant summary in the future. I think we'd better keep this.